graph-builder/src/plugins/openshift_secondary_metadata_parser/plugin.rs

```diff
@@ -0,0 +1,513 @@
+                        }
+                        _ => false,
+                    }
+                });
+            })
+        });
+
+        Ok(())
+    }
+
+    async fn process_blocked_edges(&self, io: &mut InternalIO) -> Fallible<()> {
+        let blocked_edges_dir = self.settings.data_directory.join("blocked-edges");
+        let blocked_edges: Vec<graph_data_model::BlockedEdge> =
+            deserialize_directory_files(&blocked_edges_dir, regex::Regex::new("ya+ml")?)
+                .await
+                .context(format!(
+                    "Reading blocked edges from {:?}",
+                    blocked_edges_dir
+                ))?;
+        debug!(
+            "Found {} valid blocked edges declarations.",
+            blocked_edges.len()
+        );
+
+        blocked_edges
+            .into_iter()
+            .try_for_each(|blocked_edge| -> Fallible<()> {
+                let mut to = blocked_edge.to;
+
+                // add build information to match architecture
+                if to.build.is_empty() {
+                    let arch =
+                        // Special case for a few cases where the "s390x" arch was encoded with '-' instead of '+'
+                        if to.pre == vec![semver::Identifier::AlphaNumeric("s390x".to_string())] {
+                            "s390x"
+                        } else {
+                            &self.settings.default_arch
+                        };
+
+                    warn!("Adding architecture {} to {:?}", &arch, &to);
+                    to.build
+                        .push(semver::Identifier::AlphaNumeric(arch.to_string()));
+                };
+
+                // find version in the graph
+                let release_id = match io.graph.find_by_version(&to.to_string()) {
+                    Some(release_id) => release_id,
+                    None => {
+                        warn!("Release with version {} not found in graph", to);
+                        return Ok(());
+                    }
+                };
+
+                // add metadata to block edge using the `previous.remove_regex` metadata
+                io.graph
+                    .get_metadata_as_ref_mut(&release_id)
+                    .context(format!(
+                        "[blocked_edges] Getting mutable metadata for {}",
+                        &to.to_string()
+                    ))?
+                    .insert(
+                        format!("{}.{}", self.settings.key_prefix, "previous.remove_regex"),
+                        blocked_edge.from.to_string(),
+                    );
+
+                Ok(())
+            })?;
+
+        Ok(())
+    }
+
+    async fn process_channels(&self, io: &mut InternalIO) -> Fallible<()> {
+        let channels_dir = self.settings.data_directory.join("channels");
+        let channels: Vec<graph_data_model::Channel> =
+            deserialize_directory_files(&channels_dir, regex::Regex::new("ya+ml")?)
```

Comment:
Perhaps its worth sticking to `yaml` only until file extensions go out of control :)

Reply:
This is currently in line with #226. I think both are common extensions. Do you think we should make it configurable?

Reply:
They are common, but I'd rather not leave a space for errors here.

@wking @LalatenduMohanty do you agree we should stick to "<file>.yaml" and ignore (possible) "<file>.yml" in graph-data repo?

Reply:
Standard practice for yaml file uses  `.yaml , .yml` . So we should only support these two extensions. Just supporting `yaml` is fine too but then someone might request to add `yml` as well.  So supporting `.yaml , .yml` makes us future proof.